eslint.config.mjs

```diff
@@ -14,6 +15,9 @@ export default tseslint.config(
   {
     files: ['**/*.ts', '**/*.tsx'],
     rules: {
+      // React hooks rules
+      'react-hooks/rules-of-hooks': 'warn',
```

Comment:
Should be error but then we need to fix all errors related with hooks first and this should be its own pr